frontend/src/pages/connectionTypes/manage/advanced/UriAdvancedPropertiesForm.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { Button, FormGroup } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { UriField } from '~/concepts/connectionTypes/types';
+import {
+  URI_SCHEME_REGEX,
+  isDuplicateExtension,
+} from '~/concepts/connectionTypes/fields/fieldUtils';
+import { AdvancedFieldProps } from '~/pages/connectionTypes/manage/advanced/types';
+import ExpandableFormSection from '~/components/ExpandableFormSection';
+import FieldRegexValidationRow from './FieldRegexValidationRow';
+
+const UriAdvancedPropertiesForm: React.FC<AdvancedFieldProps<UriField>> = ({
+  properties,
+  onChange,
+  onValidate,
+}) => {
+  const displayedSchemes = React.useMemo(
+    () => [...(properties.schemes?.length ? properties.schemes : [''])],
+    [properties.schemes],
+  );
+  const lastTextRef = React.useRef<HTMLInputElement | null>(null);
+
+  React.useEffect(() => {
+    if (!properties.schemes?.length) {
+      onValidate(true);
+      return;
+    }
+    const valid = properties.schemes.every(
+      (extension, i) =>
+        !extension ||
+        (URI_SCHEME_REGEX.test(extension) && !isDuplicateExtension(i, properties.schemes || [])),
+    );
+    onValidate(valid);
+    // do not run when callback changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [properties.schemes]);
+
+  return (
+    <ExpandableFormSection
+      toggleText="Advanced settings"
+      initExpanded={!!properties.schemes?.length}
+      data-testid="advanced-settings-toggle"
+    >
+      <FormGroup label="Allow specific schemes" fieldId="scheme-types" isStack>
+        {displayedSchemes.map((scheme, index) => (
+          <FieldRegexValidationRow
+            key={index}
+            id="scheme-types"
+            value={scheme}
+            isDuplicate={isDuplicateExtension(index, displayedSchemes)}
+            textRef={index === displayedSchemes.length - 1 ? lastTextRef : undefined}
+            onChange={(val) => {
+              if (!val && displayedSchemes.length === 1) {
+                onChange({ ...properties, schemes: [] });
+              } else {
+                onChange({
+                  ...properties,
+                  schemes: [
+                    ...displayedSchemes.slice(0, index),
+                    val,
+                    ...displayedSchemes.slice(index + 1),
+                  ],
```

Comment:
You could use `splice` instead
```suggestion
                  schemes: displayedSchemes.splice(index, 0, val)
```

Reply:
So `splice(...)` mutates the original array, which can cause unintended side effects. So that's why  it wasn't originally used i think (I copy pasted a lot of it)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Interestingly it looks like there is a new function called `toSpliced(...)` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced but that seems too new and might not work on older browsers